src/policy/fees.cpp

```diff
@@ -291,7 +291,7 @@ double TxConfirmStats::EstimateMedianVal(int confTarget, double sufficientTxVal,
         totalNum += txCtAvg[bucket];
         failNum += failAvg[periodTarget - 1][bucket];
         for (unsigned int confct = confTarget; confct < GetMaxConfirms(); confct++)
-            extraNum += unconfTxs[(nBlockHeight - confct)%bins][bucket];
+            extraNum += unconfTxs[(nBlockHeight >= confct ? nBlockHeight - confct : 0) % bins][bucket];
```

Comment:
This change needs extra review. Is this the correct way to handle this wrap-around?

Reply:
Does it matters if `nBlockHeight < GetMaxConfirms()`? Maybe we should just skip `EstimateMedianVal` entirely if so. cc @morcos

Reply:
Actually I think the existing code is broken, we should fix it by replacing the for loop with:
`for (unsigned int confct = confTarget; confct < GetMaxConfirms() && confct <= nBlockHeight; confct++)`

Reply:
That makes sense.

Reply:
Thanks @promag and @morcos. Now updated. Please re-review :-)